### ISSUE TITLE
"iI" command enhancement

### ISSUE THREAD

**Issue by Maijin** [rizinorg/rizin#470]
**Is your feature request related to a problem? Please describe.**

* Field names in `iI` are source of confusion between users because very short;
* Fields are missing compare to checksec https://twitter.com/mtarral/status/1354007461552861184 / https://github.com/Wenzel/checksec.py from @Wenzel
* Fields are not ordered

**Describe the solution you'd like**

* [x] Use RTable (`librz/util/table.c`/`librz/include/rz_util/rz_table.h`) for `iI`
* [ ] Enhance field names
* [ ] Add missing fields from CheckSec for PE
* [ ] Add missing fields from CheckSec for ELF

![image](https://user-images.githubusercontent.com/203261/108956582-e8f7bd80-76aa-11eb-9ee0-4d7bc9bbb8e4.png)

**Comment by fangxlmr** [comment]
@Liyuanpei This issue is asigned to you as RSoC micro-task, so you could work on this from now on. Feel free to ask questions and send Pull Requests.

In case for anyone who is not aware about [RSoC](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/).

CC @XVilka

**Comment by XVilka** [comment]
Issue is unassigned.

**Comment by sandhrabino** [comment]
What are all the expected values in the table, here?
For eg. 
![Screenshot from 2021-05-02 20-30-16](https://user-images.githubusercontent.com/42898716/116817741-87117380-ab85-11eb-8333-114156327950.png)
Should all the values in the info dump be added to the table, there are 31 values printed out with `iI` command but fitting them all in the table might look messy
Could you please suggest what's expected of this?

**Comment by wargio** [comment]
i think it is ok, we can optimize by having like flags, being displayed for each binary, for example:
```
aslr_disabled.exe: nx,canary,seh,isolation
guard_cf_disabed.exe: nx,canary,askr,dyn_base,seh,isolation
```

**Comment by sandhrabino** [comment]
> i think it is ok, we can optimize by having like flags, being displayed for each binary, for example:
> 
> ```
> aslr_disabled.exe: nx,canary,seh,isolation
> guard_cf_disabed.exe: nx,canary,askr,dyn_base,seh,isolation
> ```

![Screenshot from 2021-05-03 03-10-37](https://user-images.githubusercontent.com/42898716/116828587-36b60800-abbd-11eb-83e8-2c5469b56148.png)
Is this the expected format? If this is, then I will make a pr and then continue with the changes over there.

**Comment by sandhrabino** [comment]
> > i think it is ok, we can optimize by having like flags, being displayed for each binary, for example:
> > ```
> > aslr_disabled.exe: nx,canary,seh,isolation
> > guard_cf_disabed.exe: nx,canary,askr,dyn_base,seh,isolation
> > ```
> 
> ![Screenshot from 2021-05-03 03-10-37](https://user-images.githubusercontent.com/42898716/116828587-36b60800-abbd-11eb-83e8-2c5469b56148.png)
> Is this the expected format? If this is, then I will make a pr and then continue with the changes over there.

I need to shorten the compiler intrp and machine fields with flags like mentioned before. But this is output unreadable on a bigger font size.

**Comment by wargio** [comment]
woah. that is very very long.
@XVilka @ret2libc @thestr4ng3r

**Comment by ret2libc** [comment]
What if we translate the table, like `i` does right now? Meaning that on each row you put each different field (e.g. arch, endian, laddr, RELOCS, etc.) and each column represents a different binfile. What do you think?

**Comment by wargio** [comment]
that would solve the huge length, but i wonder if we should assign this to a new cmd.

**Comment by ret2libc** [comment]
Ah, no probably can just be an enhancement to `i`? @Maijin ?

**Comment by sandhrabino** [comment]
> What if we translate the table, like `i` does right now? Meaning that on each row you put each different field (e.g. arch, endian, laddr, RELOCS, etc.) and each column represents a different binfile. What do you think?

@ret2libc @wargio So you are suggesting a vertical table similar to the output we get now?

**Comment by wargio** [comment]
yes

**Comment by XVilka** [comment]
Agree on the vertical table

**Comment by sandhrabino** [comment]
> Agree on the vertical table

I have implemented gotten the table printed vertically but the colours and alignment is still messed up.I Will update the progress here.
This is how it looks right now.
![Screenshot from 2021-05-05 20-10-13](https://user-images.githubusercontent.com/42898716/117160616-e3230480-adde-11eb-8457-62ae7ca67efb.png)

**Comment by wargio** [comment]
i think this output is good (besides the color issue)

**Comment by Maijin** [comment]
Make sure you implement that logic in RzTable so it can be reused later :)

**Comment by sandhrabino** [comment]
> Make sure you implement that logic in RzTable so it can be reused later :)

Yes I'm trying to implement the unimplemented API to get the transpose of a table.  n this is just being printed, have to get it in an RzTable.

**Comment by sandhrabino** [comment]
Implemented the table logic here:
```[0x00001060]> iI
KEY      VALUE
--------------
arch     x86
baddr    0x00000000
binsz    14711
bintype  elf
bin      64
class    ELF64
compiler GCC: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
crypto   false
endian   LE
havecode true
intrp    /lib64/ld-linux-x86-64.so.2
laddr    0x00000000
lang     c
linenum  true
lsyms    true
machine  AMD x86-64 architecture
maxopz   16
minopz   1
os       linux
pcalign  0
rpath    NONE
sanitiz  false
static   false
stripped false
subsys   linux
va       true
RELOCS   true
CANARY   false
NX       true
PIE      true
RELRO    full
```

**Comment by Maijin** [comment]
Great, can you do a PR for this?

**Comment by sandhrabino** [comment]
> Great, can you do a PR for this?

Sure will send one.

**Comment by XVilka** [comment]
Most of it was addressed already, could you please recheck @Maijin?

**Comment by Maijin** [comment]
Hey so I checked and unfortunately no the issue is far from being completed:

- `iIt` shows in `iI?` but doesn't work
- `iI` field names are still quite obscure and should be full not shorten
- Still a lot of checks not covered by `iI` but covered by checksec (Fortified, etc.) both for PE and ELF see README for full list

**Comment by ret2libc** [comment]
> Hey so I checked and unfortunately no the issue is far from being completed:
> 
> * `iIt` shows in `iI?` but doesn't work

It is working, it's the default mode you see when you do `iI`.

> * `iI` field names are still quite obscure and should be full not shorten

Agree.

> * Still a lot of checks not covered by `iI` but covered by checksec (Fortified, etc.) both for PE and ELF see README for full list
